Abandon in-file translations, document full-file translation guidance

duncandewhurst opened the pull request:
* Closes #229

duncandewhurst committed:
release-schema: Remove patternProperties

duncandewhurst committed:
docs/guidance/map: Add translations.md

duncandewhurst committed:
docs/schema/reference.md: Remove in-file translation content

duncandewhurst committed:
Update changelog

jpmckinney committed:
Merge branch '1.2-dev' into 229-translations

jpmckinney commented on docs/guidance/map.md:
```diff
@@ -110,6 +110,7 @@ map/organization_personal_identifiers
 map/organizational_units
 map/organization_classifications
 map/beneficial_ownership
+map/translations
```
I think we instead want a paragraph with a link to this page under "Localize OCDS to your context".

duncandewhurst replied:
Done in e88f438da8c611674d9cb7a448950d2aa4db6776

jpmckinney commented on docs/schema/reference.md:
````diff
@@ -34,35 +34,11 @@ There can be cases where a publisher needs to remove, rather than update, a valu
-:jsonpointer:
-:expand: tender,item
+```{deprecated} 1.2
+  Use [full-file translations](../guidance/map/translations.md) instead.
````
This entire section is deprecated, so I think instead delete the "Many publishers..." paragraph and:
```suggestion
OCDS 1.1 allowed data to be published in multiple languages by suffixing a language code to a field name: for example, `title` for the default language and `title_es` for Spanish. OCDS 1.2 uses [full-file translations](../guidance/map/translations), instead.
```

duncandewhurst replied:
Paragraph removed in cb37d9935cab44a24c293647f520554ad6a2ee6d.

jpmckinney committed:
manage: Fix pre-commit to make more fields nontranslatable

jpmckinney reviewed (commented):
I don't think this closes #1064, which wants to cover translations of extensions.

I'll make translations.md copy-edits directly.

We can delete examples/language_localization/language.csv and its language.json

----

I updated manage.py to make these nontranslatable: postalCode, email, telephone, faxNumber, milestone code. These have publisher-specific formats or codelists, or are missing a validation property like `"format": "email"` #881.

Comparing `i18n.csv` to the patternProperties, we've removed deprecated fields (good) and added:

* These 3 address components can indeed be translated
  * Address.locality
  * Address.region
  * Address.streetAddress
* I suppose we just missed these 3 originally:
  * Amendment.description
  * RelatedProcess.title
  * Tender.procurementMethodDetails
* These fields are new:
  * Award.statusDetails
  * Contract.statusDetails
  * Tender.statusDetails
  * Tender.exclusionGrounds

Also, in OCDS 1.0, `Organization.name` was translatable. In OCDS 1.1, `OrganizationReference.name` was not translatable. In OCDS 1.2, both are translatable. In other words, both `buyer/name` and the `.name` of the corresponding entry in the `parties` array are translatable. This is good.

jpmckinney commented:
I've pushed my edits to the new guidance page, for review.

jpmckinney committed:
guidance/map/translations: Rewrite with simpler structure

- Clarify that this page is for sources with distinct fields for each language. If a field mixes languages, this page is not relevant.
- Clarify that translated text must have the same content across translations.
- Instead of a long list of repetitive field names, summarize the patterns.

duncandewhurst committed:
Update docs/schema/reference.md

Co-authored-by: James McKinney <26463+jpmckinney@users.noreply.github.com>

duncandewhurst committed:
docs/schema/reference.md: Action review comments

duncandewhurst committed:
docs/guidance/map.md: Add link to translations guidance

duncandewhurst committed:
docs/examples/language_localization: Delete unused examples

duncandewhurst commented:
> We can delete examples/language_localization/language.csv and its language.json

Done in 37a024addef69ca648d720a0433f1ba68f524ead

The revised guidance page and the changes to `manage.py` look good to me.
